src/main/java/org/opensearch/agent/tools/IndexRoutingTool.java

```diff
@@ -0,0 +1,315 @@
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@ToolAnnotation(IndexRoutingTool.TYPE)
+public class IndexRoutingTool extends VectorDBTool {
+
+    public static final String TYPE = "IndexRoutingTool";
+
+    private static final String DEFAULT_DESCRIPTION = "Use this tool to select an appropriate index for user question, "
+        + "This tool take user plain original question as input and return list of most related indexes or `Not sure`. "
+        + "If the tool returns `Not sure`, mark it as final answer and ask Human to provide index name";
+
+    private static final String DEFAULT_PROMPT_TEMPLATE = String
+        .format(
+            Locale.ROOT,
+            "Human: %s\\nAssistant:",
+            "You are an experienced engineer in OpenSearch and ElasticSearch. \n"
+                + "\n"
+                + "Given a question, your task is to choose the relevant indexes from a list of indexes.\n"
+                + "\n"
+                + "For every index, you will be given the index mapping, followed by sample data from the index.\n"
+                + "\n"
+                + "The data format is like:\n"
+                + "\n"
+                + "index-1: Index Mappings:\n"
+                + "mappings of index-1\n"
+                + "Sample data:\n"
+                + "data from index-1\n"
+                + "---\n"
+                + "index-2: Index Mappings:\n"
+                + "mappings of index-2\n"
+                + "Sample data:\n"
+                + "data from index-2\n"
+                + "---\n"
+                + "...\n"
+                + "\n"
+                + "Now the actual index mappings and sample data begins:\n"
+                + "{summaries}\n"
+                + "\n"
+                + "-------------------\n"
+                + "\n"
+                + "Format the output as a comma-separated sequence, e.g. index-1, index-2, index-3. If no indexes \n"
+                + "appear relevant to the question, return the empty string ''.\n"
+                + "\n"
+                + "Just return the index names, nothing else. \n"
+                + "If you are not sure, just return 'Not sure', nothing else.\n"
+                + "\n"
+                + "Question: {question}\n"
+                + "Answer:"
+        );
+
+    public static final int DEFAULT_K = 5;
+    public static String EMBEDDING_MODEL_ID = "embedding_model_id";
+    public static String INFERENCE_MODEL_ID = "inference_model_id";
+    public static String PROMPT_TEMPLATE = "prompt_template";
+
+    @Setter
+    @Getter
+    private String name = TYPE;
+    @Getter
+    @Setter
+    private String description = DEFAULT_DESCRIPTION;
+
+    @Getter
+    @Setter
+    private String inferenceModelId;
+
+    private final MLClients mlClients;
+
+    @Setter
+    private String promptTemplate;
+
+    public IndexRoutingTool(
+        Client client,
+        NamedXContentRegistry xContentRegistry,
+        Integer docSize,
+        Integer k,
```

Comment:
In the python implementation, `k` is not needed anymore as we take the summary directly from the vecstore.

Reply:
I may misunderstand. What does the docSize mean? I thought that is the number of indexes to include.

Reply:
docSize is the documents to include, k is the top k from vector store, normally they are same value, but they could be different. For example, retrieve top 10 from vector store and return first 5 documents in search response. For this tool, they are same value.

Reply:
Ah, sorry, I confused myself between the two. I tested a heuristic to choose the number of documents used in the embedding between 5 and 1. Will share with you when we have our sync.